packages/dashboard-frontend/src/components/WorkspaceLogs/index.tsx

```diff
@@ -51,143 +51,178 @@ export class WorkspaceLogs extends React.PureComponent<Props, State> {
-    });
+    const podName = pod.metadata?.name;
+    const logs = props.podLogsFn(podName);
+    if (logs === undefined) {
+      return;
+    }
+    const containerLogs = logs[containerName];
+    if (containerLogs === undefined) {
+      return;
+    }
+    return containerLogs;
   }
 
-  private get terminal(): React.ReactElement {
-    const lines = this.getLines();
+  private showEmptyState() {
     return (
-      <div className={styles.consoleOutput}>
-        <div>{lines.length} lines</div>
-        <pre id={LOGS_CONTAINER_ID} tabIndex={0}>
-          {lines}
-        </pre>
-      </div>
+      <EmptyState>
+        <EmptyStateIcon icon={FileIcon} />
+        <Title headingLevel="h4" size="lg">
+          No Logs to show
+        </Title>
+        <EmptyStateBody>Logs will be shown for a starting workspace.</EmptyStateBody>
+      </EmptyState>
     );
   }
 
+  private handleExpansionToggle(isExpanded: boolean): void {
+    this.setState({ isExpanded });
+  }
+
+  private handleDownload(): void {
+    const { containerName } = this.state;
+    const workspace = this.findWorkspace(this.props);
+
+    const containerLogs = this.getContainerLogs(this.props, this.state);
+    const logs = containerLogs?.logs || '';
+
+    const name = workspace?.name || 'wksp';
+    const filename = `${name}-${containerName}.log`;
+    const anchor = document.createElement('a');
+    anchor.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(logs));
+    anchor.setAttribute('download', filename);
+    anchor.style.display = 'none';
+    document.body.appendChild(anchor);
+    anchor.click();
+    document.body.removeChild(anchor);
+  }
+
   render() {
-    const { isExpanded, isStarting, isFailed, logs } = this.state;
-    const shouldToggleNavbar = true;
-
-    if (isStarting === false && isFailed == false) {
-      return (
-        <EmptyState>
-          <EmptyStateIcon icon={FileIcon} />
-          <Title headingLevel="h4" size="lg">
-            No Logs to show
-          </Title>
-          <EmptyStateBody>Logs will be shown for a starting workspace.</EmptyStateBody>
-        </EmptyState>
-      );
+    const { isExpanded, pod } = this.state;
+
+    if (pod === undefined) {
+      return this.showEmptyState();
     }
 
+    const containerLogs = this.getContainerLogs(this.props, this.state);
+
+    const expandedStyle = isExpanded ? styles.viewerExpanded : '';
+    const defaultPadding = isExpanded ? 'noPadding' : 'padding';
+
     return (
-      <PageSection variant={PageSectionVariants.light}>
-        <div className={isExpanded ? styles.tabExpanded : ''}>
-          <WorkspaceLogsTools
-            logs={logs ? logs : []}
-            shouldToggleNavbar={shouldToggleNavbar}
-            handleExpand={isExpanded => {
-              this.setState({ isExpanded });
-            }}
-          />
-          {this.terminal}
-        </div>
+      <PageSection padding={{ default: defaultPadding }} className={expandedStyle}>
```

Comment:
@akurinnoy Could you leave just one constant for restyling: 'defaultPadding' or 'expandedStyle'?

Reply:
```
    const expandedStyle = isExpanded ? styles.viewerExpanded : '';
    const defaultPadding = isExpanded ? 'noPadding' : 'padding';
 ```

Reply:
I don't really understand your point. Can you add a suggestion on how this could be done?

Reply:
I mean to eliminate defaultPadding in favor of using the corresponding CSS rule.